README.md

```diff
@@ -364,6 +368,193 @@ With `concurrent`, all tasks continue running until they all complete _or_
 any task exits with a non-zero exit code, in which case all still alive tasks
 are killed and the Builder process exits with the error code.
 
+
+## npm Config
+
+`builder` supports `package.json` `config` properties the same way that `npm`
+does, with slight enhancements in consideration of multiple `package.json`'s
+in play.
+
+### `npm` Config Overview
+
+As a refresher, `npm` utilizes the `config` field of `package.json` to make
+"per-package" environment variables to `scripts` tasks. For example, if you
+have:
+
+```js
+{
+  "config": {
+    "my_name": "Bob"
+  },
+  "scripts": {
+    "get-name": "echo Hello, ${npm_package_config_my_name}."
+  }
+}
+```
+
+and ran:
+
+```sh
+$ npm run get-name
+Hello, Bob.
+```
+
+More documentation about how `npm` does per-package configuration is at:
+
+* https://docs.npmjs.com/files/package.json#config
+* https://docs.npmjs.com/misc/config#per-package-config-settings
+
+
+### Builder Configs
+
+In `builder`, for a single `package.json` this works essentially the same in
+the above example.
+
+```sh
+$ builder run get-name
+Hello, Bob.
+```
+
+However, `builder` has the added complexity of adding in `config` variables
+from archetypes and the environment. So the basic resolution order for a
+config environment variable is:
+
+1. Look to `npm_package_config_<VAR_NAME>=<VAR_VAL>` on command line.
+2. If not set, then use `<root>/package.json:config:<VAR_NAME>` value.
+3. If not set, then use `<archetype>/package.json:config:<VAR_NAME>` value.
+
+So, let's dive in to a slightly more complex example:
+
+```js
+// <archetype>/package.json
+{
+  "config": {
+    "my_name": "ARCH BOB"
+  },
+  "scripts": {
+    "get-name": "echo Hello, ${npm_package_config_my_name}."
+  }
+}
+
+// <root>/package.json
+{
+  "config": {
+    "my_name": "ROOT JANE"
+  }
+}
+```
+
+When we run the `builder` command, the `<root>` value overrides:
+
+```sh
+$ builder run get-name
+Hello, ROOT JANE.
+```
+
+We can inject a command line flag to override even this value:
+
+```sh
+$ npm_package_config_my_name="CLI JOE" builder run get-name
+Hello, CLI JOE.
```

Comment:
Would `npm config set [PKG_NAME]:my_name CLI JOE` work as well, as long as you match the root project's package name?